### PR TITLE
apps: Change default cipher to aes-256-cbc for req, cms and smime apps

### PR DESCRIPTION
**Pull request by adimail**
Following PR #25776 which implemented the -cipher option for the req command, this PR changes the default cipher from `des-ede3-cbc` to `aes-256-cbc` for the req, cms, and smime applications.

As suggested in [this issue](https://github.com/openssl/project/issues/909), this change updates the default encryption cipher to use a more secure algorithm instead of the legacy 3DES encryption.

##### Checklist
- [x] documentation is added or updated

I didn’t find any updates needed for the existing test cases, but please lmk if you recommend any changes.

### CODE REVIEW

File: doc/man1/openssl-req.pod.in
Comment (t8m) [inline doc/man1/openssl-req.pod.in:93]: You can remove this sentence as it just duplicates the information in the second sentence.

**Review by t8m — DISMISSED** [review]

**Comment by kroeckx** [comment]
I think this needs more documentation changes. The history section should probably indicate in which version the default changed. It should be added to the CHANGES.md file, and probably to NEWS.md too.

**Comment by adimail** [comment]
Sure, I'll update the CHANGES.md file to reflect the changes from version 3.4 to 3.5. For NEWS.md, should this update be included as a significant change or as a new feature?

**Comment by t8m** [comment]
> Sure, I'll update the CHANGES.md file to reflect the changes from version 3.4 to 3.5. For NEWS.md, should this update be included as a significant change or as a new feature?

A significant change.

**Comment by kroeckx** [comment]
As a significant change. A different default is something that people are sometimes surprised by.

File: apps/cms.c
Comment (mattcaswell) [inline apps/cms.c:826]: The no-des guard can now be removed. Here and in the other files.

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Review by mattcaswell — COMMENTED** [review]
The smime and cms man pages also need updates, e.g.

https://github.com/openssl/openssl/blob/59f5f6c73cd2e1e2bd8ef405fdb6fadf0711f639/doc/man1/openssl-smime.pod.in#L168-L176

and

https://github.com/openssl/openssl/blob/59f5f6c73cd2e1e2bd8ef405fdb6fadf0711f639/doc/man1/openssl-cms.pod.in#L407-L419

Please also update the HISTORY sections of these man pages as requested by @t8m to note what version (3.5) the default was changed in.

**Review by mattcaswell — DISMISSED** [review]
LGTM

File: doc/man1/openssl-cms.pod.in
Comment (t8m) [inline doc/man1/openssl-cms.pod.in:899]: Typo: default

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by mattcaswell — APPROVED** [review]

**Review by beldmit — APPROVED** [review]
LGTM

File: apps/cms.c
Comment (shahsb) [inline apps/cms.c:826]: I have one question - Do we need similar build flag changes for AES here?

File: doc/man1/openssl-smime.pod.in
Comment (shahsb) [inline doc/man1/openssl-smime.pod.in:171]: Should we remove the DES examples from here?

**Review by shahsb — CHANGES_REQUESTED** [review]
Changes LGTM.

Thanks for making the changes.

I have just couple of questions. Would appreciate your responses.

File: apps/cms.c
Comment (adimail) [reply apps/cms.c:826]: AES is widely supported and does not require a conditional compilation flag like OPENSSL_NO_DES.

While OPENSSL_NO_DES allows OpenSSL to be built without DES support, AES does not have an equivalent conditional flag.

Comment (adimail) [reply apps/cms.c:826]: still I'd like a feedback of maintainers.

File: doc/man1/openssl-smime.pod.in
Comment (adimail) [reply doc/man1/openssl-smime.pod.in:171]: thank you @shahsb,

you have a good point, the documentation should encourage people to use modern algorithms.

File: apps/cms.c
Comment (mattcaswell) [reply apps/cms.c:826]: Right. It is actually not possible to build without AES, and there is no `OPENSSL_NO_AES` guard.

File: doc/man1/openssl-smime.pod.in
Comment (mattcaswell) [reply doc/man1/openssl-smime.pod.in:171]: It would be nice to remove it, but since this PR has already achieved approval it is also fine to leave as is rather than going around another approval cycle.

Comment (t8m) [reply doc/man1/openssl-smime.pod.in:171]: I would also keep `-des3` as example for the backwards compatibility.  `-des` is useless though. But I agree with Matt, that changing this can be done as a separate PR in future.

**Comment by t8m** [comment]
Squashed and merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**
